File: C++/프로그래머스/1/131705. 삼총사/삼총사.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> number) {
    int answer = 0;
    for(int i = 0; i< number.size(); i++){
        for(int j = i+1; j < number.size(); j++){
            for(int r = j+1; r < number.size(); r++){
                if(number[i]+number[j]+number[r] == 0){
                    answer += 1;
                }
            }
        }
        
    }
    return answer;
}

File: Python3/프로그래머스/1/131705. 삼총사/삼총사.py
from itertools import combinations

def solution(number):
    answer = 0
    # for i in range(0,len(number)):
    #     for j in range(i+1,len(number)):
    #         for r in range(j+1,len(number)):
    #             if((number[i]+number[j]+number[r]) == 0):
    #                 answer += 1
    result = list(combinations(number,3))
    for i in result:
        if(sum(i) == 0):
            answer+=1
    
    
    return answer